contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,561 @@
+     *
+     * @return outputAmount         Amount of output tokens sent to the caller
+     */
+    function redeemExactSetForToken(
+        ISetToken _setToken,
+        IERC20 _outputToken,
+        uint256 _amountSetToken,
+        uint256 _minOutputReceive,
+        bytes[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isWhitelistedIssuanceModule(_issuanceModule)
+        external
+        nonReentrant
+        returns (uint256)
+    {
+
+        uint256 outputAmount;
+        _redeemExactSet(_setToken, _amountSetToken, _issuanceModule);
+
+        // Liquidate components for WETH and ignore _outputQuote
+        outputAmount = _sellComponentsForOutputToken(_setToken, _amountSetToken, _componentQuotes, _outputToken, _issuanceModule);
+        require(outputAmount >= _minOutputReceive, "ExchangeIssuance: INSUFFICIENT OUTPUT AMOUNT");
+
+        // Transfer sender output token
+        _outputToken.safeTransfer(msg.sender, outputAmount);
+        // Emit event
+        emit ExchangeRedeem(msg.sender, _setToken, _outputToken, _amountSetToken, outputAmount);
+        // Return output amount
+        return outputAmount;
+    }
+
+    /**
+     * Redeems an exact amount of SetTokens for ETH.
+     * The SetToken must be approved by the sender to this contract.
+     *
+     * @param _setToken             Address of the SetToken being redeemed
+     * @param _amountSetToken       Amount SetTokens to redeem
+     * @param _minEthReceive        Minimum amount of Eth to receive
+     * @param _componentQuotes      The encoded 0x transactions execute
+     *
+     * @return outputAmount         Amount of output tokens sent to the caller
+     */
+    function redeemExactSetForETH(
+        ISetToken _setToken,
+        uint256 _amountSetToken,
+        uint256 _minEthReceive,
+        bytes[] memory _componentQuotes,
+        address _issuanceModule
+    )
+        isWhitelistedIssuanceModule(_issuanceModule)
+        external
+        nonReentrant
+        returns (uint256)
+    {
+        _redeemExactSet(_setToken, _amountSetToken, _issuanceModule);
+        uint ethAmount = _sellComponentsForOutputToken(_setToken, _amountSetToken, _componentQuotes, IERC20(WETH), _issuanceModule);
+        require(ethAmount >= _minEthReceive, "ExchangeIssuance: INSUFFICIENT WETH RECEIVED");
+
+        IWETH(WETH).withdraw(ethAmount);
+        (payable(msg.sender)).sendValue(ethAmount);
+
+        emit ExchangeRedeem(msg.sender, _setToken, IERC20(ETH_ADDRESS), _amountSetToken, ethAmount);
+        return ethAmount;
+         
+    }
+    
+    /**
+     * Whitelists an issuance module
+     *
+     * @param _issuanceModule    Struct containing data on issuance module to add
+     */
+    function _addIssuanceModule(address _issuanceModule, bool _isDebtIssuanceModule) internal {
+        allowedIssuanceModules[_issuanceModule].isAllowed = true;
+        allowedIssuanceModules[_issuanceModule].isDebtIssuanceModule = _isDebtIssuanceModule;
```

Comment:
Out of curiosity, which features of DebtIssuanceModule would people be using here? Am asking just because DI is typically used for tokens with external positions (which this contract prohibits).

Reply:
Good question. 
@ncitron requested adding support for different Issuance modules (debt and basic) so maybe he can elaborate on the exact use case ?

Reply:
Right now GMI uses DIM since it is removable, unlike BasicIssuanceModule. We want to be able to support issuing GMI

Reply:
Ok! Makes sense.